scripts/ci/notify-test-result.js

```diff
@@ -102,11 +104,25 @@ async function notifyTestResults() {
 
     req.on('error', error => reject(error));
 
+    const data = {
+      testStatus,
+      testUrl: workflowUrl
+    };
+
+    if (versionOrTag) {
+      // Matches a staging version tag pattern.
+      const match = versionOrTag.match(/^(\d+.\d+.\d+)-\d+$/);
+      if (match) {
```

Comment:
Feels like we should do a length check in case match has fewer than expected matches. Accessing 1 directly without validating that it's a valid key seems dangerous

Reply:
I'm not sure that outcome is possible, if the capturing group doesn't exist in the string, it won't match at all and match will be null. I think the only outcomes are null or an array of length 2. I tried a couple of test strings in the console and can't think of a case that would return non-null but also not have a captured group, let me know if I haven't thought of a case though, regex isn't my strength.

Reply:
Sounds good I'm ok approving if you are confident